### ISSUE TITLE
CA1806 (Do not ignore method results) could benefit from [PureAttribute]

### ISSUE THREAD

**Issue by sharwell** [dotnet/roslyn-analyzers#1164]
#### Analyzer package

Example: Microsoft.CodeQuality.Analyzers

#### Analyzer

CA1806 (Do not ignore method results)

#### Repro steps

1. Declare a method which is marked with `[Pure]` and returns a value
1. Call the method, ignoring the return value

```csharp
[Pure]
public int Method()
  => 3;

public void Program()
{
  Method(); // CA1806??
}
```

#### Expected behavior

CA1806 could be reported.

#### Actual behavior

CA1806 is not reported.

:memo: The potential increase in false positives for this rule is unknown. However, the ability to use `var _ =` to explicitly ignore such return values could compensate for the higher rate of reports.
